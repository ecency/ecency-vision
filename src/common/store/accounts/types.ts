import { Authority } from "@hiveio/dhive";

export interface AccountProfile {
  about?: string;
  cover_image?: string;
  location?: string;
  name?: string;
  profile_image?: string;
  website?: string;
}

export interface AccountFollowStats {
  follower_count: number;
  following_count: number;
  account: string;
}

export interface BaseAccount {
  name: string;
  __loaded?: false;
}

export interface FullAccount {
  name: string;
  owner: Authority;
  active: Authority;
  posting: Authority;
  memo_key: string;
  post_count: number;
  created: string;
  reputation: string | number;
  json_metadata: string;
  posting_json_metadata: string;
  last_vote_time: string;
  last_post: string;
  reward_hbd_balance: string;
  reward_vesting_hive: string;
  reward_hive_balance: string;
  reward_vesting_balance: string;
  balance: string;
  vesting_shares: string;
  hbd_balance: string;
  savings_balance: string;
  savings_hbd_balance: string;
  next_vesting_withdrawal: string;
  delegated_vesting_shares: string;
  received_vesting_shares: string;
  vesting_withdraw_rate: string;
  to_withdraw: string;
  withdrawn: string;
  witness_votes: string[];
  proxy: string;
  proxied_vsf_votes: number[] | string[];
  voting_manabar: {
    current_mana: string | number;
    last_update_time: number;
  };
  voting_power: number;
  downvote_manabar: {
    current_mana: string | number;
    last_update_time: number;
  };
  profile?: AccountProfile;
  follow_stats?: AccountFollowStats;
  __loaded?: true;
}

export type Account = FullAccount | BaseAccount;

export type Accounts = Account[];

export enum ActionTypes {
  ADD = "@accounts/ADD"
}

export interface AddAction {
  type: ActionTypes.ADD;
  data: Account;
}

export type Actions = AddAction; // | .. | ..
