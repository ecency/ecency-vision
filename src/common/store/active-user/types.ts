import {Account} from "../accounts/types";

export interface UserPoints {
    points: string;
    uPoints: string;
}

export interface ActiveUser {
    username: string;
    data: Account;
    points: UserPoints;
}

export enum ActionTypes {
    LOGIN = "@active-user/LOGIN",
    LOGOUT = "@active-user/LOGOUT",
    UPDATE = "@active-user/UPDATE",
}

export interface LoginAction {
    type: ActionTypes.LOGIN;
}

export interface LogoutAction {
    type: ActionTypes.LOGOUT;
}

export interface UpdateAction {
    type: ActionTypes.UPDATE;
    data: Account;
    points: UserPoints
}

export type Actions = LoginAction | LogoutAction | UpdateAction;
