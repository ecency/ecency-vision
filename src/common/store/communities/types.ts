export enum ROLES {
  OWNER = 'owner',
  ADMIN = 'admin',
  MOD = 'mod',
  MEMBER = 'member',
  GUEST = 'guest',
  MUTED = 'muted'
}

export const roleMap: Record<string, string[]> = {
  [ROLES.OWNER]: [ROLES.ADMIN, ROLES.MOD, ROLES.MEMBER, ROLES.GUEST, ROLES.MUTED],
  [ROLES.ADMIN]: [ROLES.MOD, ROLES.MEMBER, ROLES.GUEST, ROLES.MUTED],
  [ROLES.MOD]: [ROLES.MEMBER, ROLES.GUEST, ROLES.MUTED]
};

export type CommunityTeam = Array<Array<string>>;

export interface Community {
  about: string;
  admins?: string[];
  avatar_url: string;
  created_at: string;
  description: string;
  flag_text: string;
  id: number;
  is_nsfw: boolean;
  lang: string;
  name: string;
  num_authors: number;
  num_pending: number;
  subscribers: number;
  sum_pending: number;
  settings?: any;
  team: CommunityTeam;
  title: string;
  type_id: number;
}

export type Communities = Community[];

export enum ActionTypes {
  ADD = '@communities/ADD'
}

export interface AddAction {
  type: ActionTypes.ADD;
  data: Community;
}

export type Actions = AddAction; // | .. | ..
