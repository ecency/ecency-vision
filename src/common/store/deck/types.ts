export interface SerializedDeckModel {
  dataParams: any[];
  header: {
    title: string;
    updateIntervalMs: number;
  };
  dataFilters: any;
  createdAt: Date;
}

export interface DeckModel extends SerializedDeckModel {
  data?: any;
  header: {
    title: string;
    icon: JSX.Element | null;
    reloading: boolean;
    updateIntervalMs: number;
  };
  listItemComponent: any;
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}

export interface WalletDeckModel extends IdentifiableDeckModel {
  dataFilters: {
    group: string;
  };
}

export enum ActionTypes {
  CREATE = "@deck/CREATE",
  SET_DATA = "@deck/FETCH_DATA",
  SET_RELOADING = "@deck/SET_RELOADING",
  REORDER = "@deck/REORDER",
  DELETE = "@deck/DELETE",
  DELETE_ALL = "@deck/DELETE_ALL",
  UPDATE_INTERVAL = "@deck/UPDATE_INTERVAL",
  SET_DATA_FILTERS = "@deck/SET_DATA_FILTERS"
}

export interface CreateAction {
  type: ActionTypes.CREATE;
  data: [
    DeckModel["listItemComponent"],
    DeckModel["header"]["title"],
    DeckModel["header"]["icon"],
    DeckModel["dataParams"],
    DeckModel["createdAt"]?,
    DeckModel["header"]["updateIntervalMs"]?,
    DeckModel["dataFilters"]?
  ];
}

export interface SetDataAction {
  type: ActionTypes.SET_DATA;
  data: {
    title: string;
    data: any;
  };
}

export interface SetReloadingAction {
  type: ActionTypes.SET_RELOADING;
  data: {
    title: string;
    isReloading: boolean;
  };
}

export interface ReOrderAction {
  type: ActionTypes.REORDER;
  data: {
    startIndex: number;
    endIndex: number;
  };
}

export interface DeleteAction {
  type: ActionTypes.DELETE;
  data: {
    title: string;
  };
}

export interface DeleteAllAction {
  type: ActionTypes.DELETE_ALL;
}

export interface UpdateIntervalAction {
  type: ActionTypes.UPDATE_INTERVAL;
  data: {
    title: string;
    updateIntervalMs: DeckModel["header"]["updateIntervalMs"];
  };
}

export interface SetDataFiltersAction {
  type: ActionTypes.SET_DATA_FILTERS;
  data: {
    title: string;
    username: string;
    dataFilters: any;
  };
}

export type DeckState = {
  items: IdentifiableDeckModel[];
  isContentLoading: boolean;
};

export type Actions =
  | CreateAction
  | SetDataAction
  | SetReloadingAction
  | ReOrderAction
  | DeleteAction
  | UpdateIntervalAction
  | SetDataFiltersAction
  | DeleteAllAction;
