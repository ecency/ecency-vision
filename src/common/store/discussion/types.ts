import {Entry} from '../entries/types';

export enum SortOrder {
  trending = 'trending',
  author_reputation = 'author_reputation',
  votes = 'votes',
  created = 'created',
}

export interface Discussion {
  list: Entry[];
  loading: boolean;
  error: boolean;
  order: SortOrder;
}

export enum ActionTypes {
  FETCH = '@discussions/FETCH',
  FETCHED = '@discussions/FETCHED',
  FETCH_ERROR = '@discussions/FETCH_ERROR',
  RESET = '@discussions/RESET',
  SET_ORDER = '@discussions/SET_ORDER',
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  list: Entry[];
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export interface ResetAction {
  type: ActionTypes.RESET;
}

export interface SetOrderAction {
  type: ActionTypes.SET_ORDER;
  list: Entry[];
  order: SortOrder;
}

export type Actions =
  | FetchAction
  | FetchedAction
  | FetchErrorAction
  | ResetAction
  | SetOrderAction;
