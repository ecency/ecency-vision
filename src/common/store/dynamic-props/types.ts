export interface DynamicProps {
  hivePerMVests: number;
  base: number;
  quote: number;
  fundRewardBalance: number;
  fundRecentClaims: number;
  hbdPrintRate: number;
  hbdInterestRate: number;
  headBlock: number;
  totalVestingFund: number;
  totalVestingShares: number;
  virtualSupply: number;
  vestingRewardPercent: number;
}

export type State = DynamicProps;

export enum ActionTypes {
  FETCHED = "@dynamic-props/FETCHED"
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  props: DynamicProps;
}

export type Actions = FetchedAction; // |..|..Ã¦
