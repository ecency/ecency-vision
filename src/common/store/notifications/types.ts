import { LoginAction, LogoutAction } from '../active-user/types';

// Web socket notification types

interface BaseWsNotification {
  source: string;
  target: string;
  timestamp: string;
}

export interface WsVoteNotification extends BaseWsNotification {
  type: 'vote';
  extra: {
    permlink: string;
    weight: number;
    title: string | null;
    img_url: string | null;
  };
}

export interface WsMentionNotification extends BaseWsNotification {
  type: 'mention';
  extra: {
    permlink: string;
    is_post: 0 | 1;
    title: string | null;
    img_url: string | null;
  };
}

export interface WsFollowNotification extends BaseWsNotification {
  type: 'follow';
  extra: {
    what: string[];
  };
}

export interface WsReplyNotification extends BaseWsNotification {
  type: 'reply';
  extra: {
    title: string;
    body: string;
    json_metadata: string;
    permlink: string;
    parent_author: string;
    parent_permlink: string;
    parent_title: string | null;
    parent_img_url: string | null;
  };
}

export interface WsReblogNotification extends BaseWsNotification {
  type: 'reblog';
  extra: {
    permlink: string;
    title: string | null;
    img_url: string | null;
  };
}

export interface WsTransferNotification extends BaseWsNotification {
  type: 'transfer';
  extra: {
    amount: string;
    memo: string;
  };
}

export interface WsSpinNotification extends BaseWsNotification {
  type: 'spin';
}

export interface WsInactiveNotification extends BaseWsNotification {
  type: 'inactive';
}

export interface WsReferralNotification extends BaseWsNotification {
  type: 'referral';
}

export type WsNotification =
  | WsVoteNotification
  | WsMentionNotification
  | WsFollowNotification
  | WsReplyNotification
  | WsReblogNotification
  | WsTransferNotification
  | WsSpinNotification
  | WsInactiveNotification
  | WsReferralNotification;

// HTTP api notification types

interface BaseAPiNotification {
  id: string;
  source: string;
  read: 0 | 1;
  timestamp: string; // iso formatted date
  ts: number; // unix timestamp
  gk: string; // group key
  gkf: boolean; // group key flag. true when a new group started
}

export interface ApiVoteNotification extends BaseAPiNotification {
  type: 'vote' | 'unvote';
  voter: string;
  weight: number;
  author: string;
  permlink: string;
  title: string | null;
  img_url: string | null;
}

export interface ApiMentionNotification extends BaseAPiNotification {
  type: 'mention';
  author: string;
  account: string;
  permlink: string;
  post: boolean;
  title: string | null;
  img_url: string | null;
}

export interface ApiFollowNotification extends BaseAPiNotification {
  type: 'follow' | 'unfollow' | 'ignore';
  follower: string;
  following: string;
  blog: boolean;
}

export interface ApiReblogNotification extends BaseAPiNotification {
  type: 'reblog';
  account: string;
  author: string;
  permlink: string;
  title: string | null;
  img_url: string | null;
}

export interface ApiReplyNotification extends BaseAPiNotification {
  type: 'reply';
  author: string;
  permlink: string;
  title: string;
  body: string;
  json_metadata: string;
  metadata: any;
  parent_author: string;
  parent_permlink: string;
  parent_title: string | null;
  parent_img_url: string | null;
}

export interface ApiTransferNotification extends BaseAPiNotification {
  type: 'transfer';
  to: string;
  amount: string;
  memo: string | null;
}

export interface ApiSpinNotification extends BaseAPiNotification {
  type: 'spin';
}

export interface ApiInactiveNotification extends BaseAPiNotification {
  type: 'inactive';
}

export interface ApiReferralNotification extends BaseAPiNotification {
  type: 'referral';
}

export type ApiNotification =
  | ApiVoteNotification
  | ApiMentionNotification
  | ApiFollowNotification
  | ApiReblogNotification
  | ApiReplyNotification
  | ApiTransferNotification
  | ApiSpinNotification
  | ApiInactiveNotification
  | ApiReferralNotification;

export enum NotificationFilter {
  VOTES = 'rvotes',
  MENTIONS = 'mentions',
  FOLLOWS = 'follows',
  REPLIES = 'replies',
  REBLOGS = 'reblogs',
  TRANSFERS = 'transfers'
}

export enum NFetchMode {
  REPLACE = 'REPLACE',
  APPEND = 'APPEND'
}

export interface Notifications {
  filter: NotificationFilter | null;
  unread: number;
  list: ApiNotification[];
  loading: boolean;
  hasMore: boolean;
  unreadFetchFlag: boolean;
}

export enum ActionTypes {
  FETCH = '@notifications/FETCH',
  FETCHED = '@notifications/FETCHED',
  SET_FILTER = '@notifications/SET_FILTER',
  SET_UNREAD_COUNT = '@notifications/SET_UNREAD_COUNT',
  MARK = '@notifications/MARK'
}

export interface FetchAction {
  type: ActionTypes.FETCH;
  mode: NFetchMode;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  list: ApiNotification[];
  mode: NFetchMode;
}

export interface SetFilterAction {
  type: ActionTypes.SET_FILTER;
  filter: NotificationFilter | null;
}

export interface SetUnreadCountAction {
  type: ActionTypes.SET_UNREAD_COUNT;
  count: number;
}

export interface MarkAction {
  type: ActionTypes.MARK;
  id: string | null;
}

export type Actions =
  | FetchAction
  | FetchedAction
  | SetFilterAction
  | SetUnreadCountAction
  | LoginAction
  | LogoutAction
  | MarkAction;
