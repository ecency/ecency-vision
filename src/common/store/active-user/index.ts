import {Dispatch} from 'redux';

import Cookies from 'js-cookie';

import {AppState} from '../index';
import {Account} from '../accounts/types';
import {
  Actions,
  ActionTypes,
  ActiveUser,
  UserPoints,
  LoginAction,
  LogoutAction,
  UpdateAction,
} from './types';

import * as ls from '../../util/local-storage';

import {getAccount} from '../../api/hive';
import {getPoints} from '../../api/private-api';

import {activeUserMaker} from '../helper';

const load = (): ActiveUser | null => {
  const name = ls.get('active_user');
  if (name && ls.get(`user_${name}`)) {
    return activeUserMaker(name);
  }

  return null;
};

export const initialState: ActiveUser | null = null;

export default (
  state: ActiveUser | null = initialState,
  action: Actions,
): ActiveUser | null => {
  switch (action.type) {
    case ActionTypes.LOGIN:
    case ActionTypes.LOGOUT: {
      return load();
    }
    case ActionTypes.UPDATE: {
      const {data, points} = action;
      return Object.assign({}, state, {data, points});
    }
    default:
      return state;
  }
};

/* Actions */
export const setActiveUser =
  (name: string | null) => async (dispatch: Dispatch) => {
    if (name) {
      ls.set('active_user', name);
      Cookies.set('active_user', name);
      dispatch(loginAct());
    } else {
      ls.remove('active_user');
      Cookies.remove('active_user');
      dispatch(logoutAct());
    }
  };

export const updateActiveUser =
  (data?: Account) => async (dispatch: Dispatch, getState: () => AppState) => {
    const {activeUser} = getState();
    if (!activeUser) {
      return;
    }

    let uData: Account | undefined = data;
    if (!uData) {
      try {
        uData = await getAccount(activeUser.username);
      } catch (e) {
        uData = {
          name: activeUser.username,
        };
      }
    }

    let points: UserPoints;
    try {
      const r = await getPoints(activeUser.username);
      points = {
        points: r.points,
        uPoints: r.unclaimed_points,
      };
    } catch (e) {
      points = {
        points: '0.000',
        uPoints: '0.000',
      };
    }

    dispatch(updateAct(uData, points));
  };

/* Action Creators */
export const loginAct = (): LoginAction => {
  return {
    type: ActionTypes.LOGIN,
  };
};

export const logoutAct = (): LogoutAction => {
  return {
    type: ActionTypes.LOGOUT,
  };
};

export const updateAct = (data: Account, points: UserPoints): UpdateAction => {
  return {
    type: ActionTypes.UPDATE,
    data,
    points,
  };
};
