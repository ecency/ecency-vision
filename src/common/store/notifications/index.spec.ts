import reducer, {
  initialState,
  fetchAct,
  fetchedAct,
  setFilterAct,
  setUnreadCountAct,
  markAct,
} from './index';
import {NFetchMode, NotificationFilter} from './types';
import {
  apiNotificationList1,
  apiNotificationList2,
} from '../../helper/test-helper';

import {logoutAct} from '../active-user';

let state = initialState;

it('1- default state', () => {
  expect(state).toMatchSnapshot();
});

it('2- fetchAct', () => {
  state = reducer(state, fetchAct(NFetchMode.REPLACE));
  expect(state).toMatchSnapshot();
});

it('3- fetchedAct', () => {
  state = reducer(state, fetchedAct(apiNotificationList1, NFetchMode.REPLACE));
  expect(state).toMatchSnapshot();
});

it('4- fetchedAct', () => {
  state = reducer(state, fetchedAct(apiNotificationList2, NFetchMode.APPEND));
  expect(state).toMatchSnapshot();
});

it('5- markAct', () => {
  state = reducer(state, markAct('m-49590315'));
  expect(state).toMatchSnapshot();
});

it('6- markAct', () => {
  state = reducer(state, markAct(null));
  expect(state).toMatchSnapshot();
});

it('8- setFilterAct', () => {
  state = reducer(state, setFilterAct(NotificationFilter.VOTES));
  expect(state).toMatchSnapshot();
});

it('9- setUnreadCountAct', () => {
  state = reducer(state, setUnreadCountAct(8));
  expect(state).toMatchSnapshot();
});

it('10- logoutAct', () => {
  state = reducer(state, logoutAct());
  expect(state).toMatchSnapshot();
});
