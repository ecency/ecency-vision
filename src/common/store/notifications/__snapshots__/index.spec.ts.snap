// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "fbSupport": "pending",
  "filter": null,
  "hasMore": true,
  "list": Array [],
  "loading": false,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;

exports[`2- fetchAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": null,
  "hasMore": true,
  "list": Array [],
  "loading": true,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;

exports[`3- fetchedAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": null,
  "hasMore": false,
  "list": Array [
    Object {
      "author": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "v-705431149",
      "img_url": null,
      "permlink": "update-on-hivesigner-integration-into-condenser",
      "read": 0,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:50:48+00:00",
      "title": "Update on Hivesigner integration into Condenser",
      "ts": 1595566248,
      "type": "vote",
      "voter": "fooo10",
      "weight": 10000,
    },
    Object {
      "account": "good-karma",
      "author": "ecency.stats",
      "gk": "8 hours",
      "gkf": true,
      "id": "m-49590315",
      "img_url": "https://images.ecency.com/0x0/https://img.esteem.app/08zx2x.jpg",
      "permlink": "daily-activities-points-2020-07-24",
      "post": true,
      "read": 0,
      "source": "ecency.stats",
      "timestamp": "2020-07-24T01:10:09+00:00",
      "title": "Daily activities [Points] - Jul 24",
      "ts": 1595545809,
      "type": "mention",
    },
    Object {
      "blog": true,
      "follower": "fooo10",
      "following": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "f-174272007",
      "read": 0,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:08:24+00:00",
      "ts": 1595563704,
      "type": "follow",
    },
    Object {
      "blog": false,
      "follower": "fooo102",
      "following": "good-karma",
      "gk": "2020-07-19",
      "gkf": false,
      "id": "f-174258147",
      "read": 1,
      "source": "fooo102",
      "timestamp": "2020-07-19T05:07:09+00:00",
      "ts": 1595128029,
      "type": "unfollow",
    },
  ],
  "loading": false,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;

exports[`4- fetchedAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": null,
  "hasMore": false,
  "list": Array [
    Object {
      "author": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "v-705431149",
      "img_url": null,
      "permlink": "update-on-hivesigner-integration-into-condenser",
      "read": 0,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:50:48+00:00",
      "title": "Update on Hivesigner integration into Condenser",
      "ts": 1595566248,
      "type": "vote",
      "voter": "fooo10",
      "weight": 10000,
    },
    Object {
      "account": "good-karma",
      "author": "ecency.stats",
      "gk": "8 hours",
      "gkf": true,
      "id": "m-49590315",
      "img_url": "https://images.ecency.com/0x0/https://img.esteem.app/08zx2x.jpg",
      "permlink": "daily-activities-points-2020-07-24",
      "post": true,
      "read": 0,
      "source": "ecency.stats",
      "timestamp": "2020-07-24T01:10:09+00:00",
      "title": "Daily activities [Points] - Jul 24",
      "ts": 1595545809,
      "type": "mention",
    },
    Object {
      "blog": true,
      "follower": "fooo10",
      "following": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "f-174272007",
      "read": 0,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:08:24+00:00",
      "ts": 1595563704,
      "type": "follow",
    },
    Object {
      "blog": false,
      "follower": "fooo102",
      "following": "good-karma",
      "gk": "2020-07-19",
      "gkf": false,
      "id": "f-174258147",
      "read": 1,
      "source": "fooo102",
      "timestamp": "2020-07-19T05:07:09+00:00",
      "ts": 1595128029,
      "type": "unfollow",
    },
    Object {
      "author": "dunsky",
      "body": "Testing replies from Ecency web interface ðŸ˜Ž",
      "gk": "2020-07-21",
      "gkf": false,
      "id": "rr-108155773",
      "json_metadata": "{\\"tags\\": [\\"ecency\\"], \\"app\\": \\"ecency/0.0.1-vision\\", \\"format\\": \\"markdown+html\\"}",
      "metadata": Object {
        "app": "ecency/0.0.1-vision",
        "format": "markdown+html",
        "tags": Array [
          "ecency",
        ],
      },
      "parent_author": "good-karma",
      "parent_img_url": null,
      "parent_permlink": "re-dunsky-2020713t164324912z",
      "parent_title": null,
      "permlink": "re-good-karma-2020721t1915968z",
      "read": 1,
      "source": "dunsky",
      "timestamp": "2020-07-21T16:02:00+00:00",
      "title": "",
      "ts": 1595340120,
      "type": "reply",
    },
    Object {
      "account": "foo220",
      "author": "good-karma",
      "gk": "Yesterday",
      "gkf": true,
      "id": "r-10661404",
      "img_url": null,
      "permlink": "update-on-hivesigner-integration-into-condenser",
      "read": 0,
      "source": "foo220",
      "timestamp": "2020-07-23T13:36:06+00:00",
      "title": "Update on Hivesigner integration into Condenser",
      "ts": 1595504166,
      "type": "reblog",
    },
    Object {
      "amount": "900.000 HBD",
      "gk": "2020-07-20",
      "gkf": true,
      "id": "t-62284775",
      "memo": "",
      "read": 1,
      "source": "bar212",
      "timestamp": "2020-07-20T09:30:21+00:00",
      "to": "good-karma",
      "ts": 1595230221,
      "type": "transfer",
    },
  ],
  "loading": false,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;

exports[`5- markAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": null,
  "hasMore": false,
  "list": Array [
    Object {
      "author": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "v-705431149",
      "img_url": null,
      "permlink": "update-on-hivesigner-integration-into-condenser",
      "read": 0,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:50:48+00:00",
      "title": "Update on Hivesigner integration into Condenser",
      "ts": 1595566248,
      "type": "vote",
      "voter": "fooo10",
      "weight": 10000,
    },
    Object {
      "account": "good-karma",
      "author": "ecency.stats",
      "gk": "8 hours",
      "gkf": true,
      "id": "m-49590315",
      "img_url": "https://images.ecency.com/0x0/https://img.esteem.app/08zx2x.jpg",
      "permlink": "daily-activities-points-2020-07-24",
      "post": true,
      "read": 1,
      "source": "ecency.stats",
      "timestamp": "2020-07-24T01:10:09+00:00",
      "title": "Daily activities [Points] - Jul 24",
      "ts": 1595545809,
      "type": "mention",
    },
    Object {
      "blog": true,
      "follower": "fooo10",
      "following": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "f-174272007",
      "read": 0,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:08:24+00:00",
      "ts": 1595563704,
      "type": "follow",
    },
    Object {
      "blog": false,
      "follower": "fooo102",
      "following": "good-karma",
      "gk": "2020-07-19",
      "gkf": false,
      "id": "f-174258147",
      "read": 1,
      "source": "fooo102",
      "timestamp": "2020-07-19T05:07:09+00:00",
      "ts": 1595128029,
      "type": "unfollow",
    },
    Object {
      "author": "dunsky",
      "body": "Testing replies from Ecency web interface ðŸ˜Ž",
      "gk": "2020-07-21",
      "gkf": false,
      "id": "rr-108155773",
      "json_metadata": "{\\"tags\\": [\\"ecency\\"], \\"app\\": \\"ecency/0.0.1-vision\\", \\"format\\": \\"markdown+html\\"}",
      "metadata": Object {
        "app": "ecency/0.0.1-vision",
        "format": "markdown+html",
        "tags": Array [
          "ecency",
        ],
      },
      "parent_author": "good-karma",
      "parent_img_url": null,
      "parent_permlink": "re-dunsky-2020713t164324912z",
      "parent_title": null,
      "permlink": "re-good-karma-2020721t1915968z",
      "read": 1,
      "source": "dunsky",
      "timestamp": "2020-07-21T16:02:00+00:00",
      "title": "",
      "ts": 1595340120,
      "type": "reply",
    },
    Object {
      "account": "foo220",
      "author": "good-karma",
      "gk": "Yesterday",
      "gkf": true,
      "id": "r-10661404",
      "img_url": null,
      "permlink": "update-on-hivesigner-integration-into-condenser",
      "read": 0,
      "source": "foo220",
      "timestamp": "2020-07-23T13:36:06+00:00",
      "title": "Update on Hivesigner integration into Condenser",
      "ts": 1595504166,
      "type": "reblog",
    },
    Object {
      "amount": "900.000 HBD",
      "gk": "2020-07-20",
      "gkf": true,
      "id": "t-62284775",
      "memo": "",
      "read": 1,
      "source": "bar212",
      "timestamp": "2020-07-20T09:30:21+00:00",
      "to": "good-karma",
      "ts": 1595230221,
      "type": "transfer",
    },
  ],
  "loading": false,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;

exports[`6- markAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": null,
  "hasMore": false,
  "list": Array [
    Object {
      "author": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "v-705431149",
      "img_url": null,
      "permlink": "update-on-hivesigner-integration-into-condenser",
      "read": 1,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:50:48+00:00",
      "title": "Update on Hivesigner integration into Condenser",
      "ts": 1595566248,
      "type": "vote",
      "voter": "fooo10",
      "weight": 10000,
    },
    Object {
      "account": "good-karma",
      "author": "ecency.stats",
      "gk": "8 hours",
      "gkf": true,
      "id": "m-49590315",
      "img_url": "https://images.ecency.com/0x0/https://img.esteem.app/08zx2x.jpg",
      "permlink": "daily-activities-points-2020-07-24",
      "post": true,
      "read": 1,
      "source": "ecency.stats",
      "timestamp": "2020-07-24T01:10:09+00:00",
      "title": "Daily activities [Points] - Jul 24",
      "ts": 1595545809,
      "type": "mention",
    },
    Object {
      "blog": true,
      "follower": "fooo10",
      "following": "good-karma",
      "gk": "2 hours",
      "gkf": true,
      "id": "f-174272007",
      "read": 1,
      "source": "fooo10",
      "timestamp": "2020-07-24T06:08:24+00:00",
      "ts": 1595563704,
      "type": "follow",
    },
    Object {
      "blog": false,
      "follower": "fooo102",
      "following": "good-karma",
      "gk": "2020-07-19",
      "gkf": false,
      "id": "f-174258147",
      "read": 1,
      "source": "fooo102",
      "timestamp": "2020-07-19T05:07:09+00:00",
      "ts": 1595128029,
      "type": "unfollow",
    },
    Object {
      "author": "dunsky",
      "body": "Testing replies from Ecency web interface ðŸ˜Ž",
      "gk": "2020-07-21",
      "gkf": false,
      "id": "rr-108155773",
      "json_metadata": "{\\"tags\\": [\\"ecency\\"], \\"app\\": \\"ecency/0.0.1-vision\\", \\"format\\": \\"markdown+html\\"}",
      "metadata": Object {
        "app": "ecency/0.0.1-vision",
        "format": "markdown+html",
        "tags": Array [
          "ecency",
        ],
      },
      "parent_author": "good-karma",
      "parent_img_url": null,
      "parent_permlink": "re-dunsky-2020713t164324912z",
      "parent_title": null,
      "permlink": "re-good-karma-2020721t1915968z",
      "read": 1,
      "source": "dunsky",
      "timestamp": "2020-07-21T16:02:00+00:00",
      "title": "",
      "ts": 1595340120,
      "type": "reply",
    },
    Object {
      "account": "foo220",
      "author": "good-karma",
      "gk": "Yesterday",
      "gkf": true,
      "id": "r-10661404",
      "img_url": null,
      "permlink": "update-on-hivesigner-integration-into-condenser",
      "read": 1,
      "source": "foo220",
      "timestamp": "2020-07-23T13:36:06+00:00",
      "title": "Update on Hivesigner integration into Condenser",
      "ts": 1595504166,
      "type": "reblog",
    },
    Object {
      "amount": "900.000 HBD",
      "gk": "2020-07-20",
      "gkf": true,
      "id": "t-62284775",
      "memo": "",
      "read": 1,
      "source": "bar212",
      "timestamp": "2020-07-20T09:30:21+00:00",
      "to": "good-karma",
      "ts": 1595230221,
      "type": "transfer",
    },
  ],
  "loading": false,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;

exports[`8- setFilterAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": "rvotes",
  "hasMore": true,
  "list": Array [],
  "loading": false,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;

exports[`9- setUnreadCountAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": "rvotes",
  "hasMore": true,
  "list": Array [],
  "loading": false,
  "unread": 8,
  "unreadFetchFlag": false,
}
`;

exports[`10- logoutAct 1`] = `
Object {
  "fbSupport": "pending",
  "filter": null,
  "hasMore": true,
  "list": Array [],
  "loading": false,
  "unread": 0,
  "unreadFetchFlag": true,
}
`;
